pkg/controller/gardenletdeployer/actuator_test.go

```diff
@@ -574,57 +535,51 @@ var _ = Describe("Interface", func() {
 				))
 			})
 
-			When("DoNotCopyBackupCredentials feature gate is enabled", func() {
-				BeforeEach(func() {
-					DeferCleanup(test.WithFeatureGate(features.DefaultFeatureGate, features.DoNotCopyBackupCredentials, true))
```

Comment:
As the suite, no longer depends on feature gates, we can clean up the registration of feature gates for it.

Suggestion:
```diff
diff --git a/pkg/controller/gardenletdeployer/gardenletdeployer_suite_test.go b/pkg/controller/gardenletdeployer/gardenletdeployer_suite_test.go
index ba7c193b14..ddec1f173f 100644
--- a/pkg/controller/gardenletdeployer/gardenletdeployer_suite_test.go
+++ b/pkg/controller/gardenletdeployer/gardenletdeployer_suite_test.go
@@ -9,12 +9,9 @@ import (
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
-
-	"github.com/gardener/gardener/pkg/gardenlet/features"
 )
 
 func TestGardenletDeployer(t *testing.T) {
-	features.RegisterFeatureGates()
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "Controller GardenletDeployer Suite")
 }

```